src/NzbDrone.Common/Http/HttpRequest.cs

```diff
@@ -54,6 +54,7 @@ public Uri Url
         public Dictionary<string, string> Cookies { get; private set; }
         public bool StoreResponseCookie { get; set; }
         public TimeSpan RateLimit { get; set; }
+        public HttpRequestProxySettings Proxy {get; set;}
```

Comment:
@markus101 Any idea on how to eliminate this property on HttpRequest? It's done this way because the settings are only available in NzbDrone.Core.
My idea atm is to define an interface in NzbDrone.Common `IHttpProxySettingsProvider.GetProxySettings(HttpRequest request)`, but provide an implementation for it in NzbDrone.Core.
The provider would be responsible for checking whether a proxy is needed, and supply the settings if so.